gobblin-restli/gobblin-job-config-service/server/src/main/java/gobblin/service/JobConfigsResource.java

```diff
@@ -0,0 +1,231 @@
+import com.linkedin.restli.common.EmptyRecord;
+import com.linkedin.restli.common.HttpStatus;
+import com.linkedin.restli.server.RestLiServiceException;
+import com.linkedin.restli.server.UpdateResponse;
+import com.linkedin.restli.server.annotations.RestLiCollection;
+import com.linkedin.restli.server.CreateResponse;
+import com.linkedin.data.template.StringMap;
+import com.linkedin.restli.server.resources.ComplexKeyResourceTemplate;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import gobblin.config.ConfigBuilder;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.runtime.api.JobSpec;
+import gobblin.runtime.api.JobSpecNotFoundException;
+import gobblin.runtime.api.MutableJobCatalog;
+import gobblin.runtime.job_catalog.FSJobCatalog;
+import gobblin.runtime.job_catalog.ImmutableFSJobCatalog;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@RestLiCollection(name = "jobconfigs", namespace = "gobblin.service")
+/**
+ * Rest.li resource for handling job configuration requests
+ */
+public class JobConfigsResource extends ComplexKeyResourceTemplate<JobConfigId, EmptyRecord, JobConfig> {
+  private static final Logger LOG = LoggerFactory.getLogger(JobConfigsResource.class);
+
+  private MutableJobCatalog jobCatalog;
+
+  JobConfigsResource(FSJobCatalog jobCatalog) {
+    this.jobCatalog = jobCatalog;
+  }
+
+  /**
+   * Logs message and throws Rest.li exception
+   * @param status HTTP status code
+   * @param msg error message
+   * @param e exception
+   */
+  public void logAndThrowRestLiServiceException(HttpStatus status, String msg, Exception e) {
+    if (e != null) {
+      LOG.error(msg, e);
+      throw new RestLiServiceException(status, msg + " cause = " + e.getMessage());
+    } else {
+      LOG.error(msg);
+      throw new RestLiServiceException(status, msg);
+    }
+  }
+
+  /**
+   * Retrieve the {@link JobConfig} with the given key
+   * @param key job config id key containing group name and job name
+   * @return {@link JobConfig} with job configuration
+   */
+  @Override
+  public JobConfig get(ComplexResourceKey<JobConfigId, EmptyRecord> key) {
+    String jobGroup = key.getKey().getJobGroup();
+    String jobName = key.getKey().getJobName();
+
+    LOG.info("Get called with jobGroup " + jobGroup + " jobName " + jobName);
+
+    try {
+      JobSpec jobSpec = jobCatalog.getJobSpec(new URI("/" + jobGroup  + "/" + jobName));
+      Properties jobProps = jobSpec.getConfigAsProperties();
+      String schedule = jobProps.getProperty(ConfigurationKeys.JOB_SCHEDULE_KEY);
+      String template = jobProps.getProperty(ConfigurationKeys.JOB_TEMPLATE_PATH);
+      // remove keys that were injected by jobCatalog or as part of jobSpec creation
+      jobProps.remove(ConfigurationKeys.JOB_SCHEDULE_KEY);
+      jobProps.remove(ConfigurationKeys.JOB_NAME_KEY);
```

Comment:
I'm not sure there's a point to removing this. I think you're trying to make it so that doing a put followed by a get returns exactly what you put in, but why is this required?